src/usr/local/buildpack/utils/filesystem.sh

```diff
@@ -25,7 +25,16 @@ function find_versioned_tool_path () {
 
 function create_versioned_tool_path () {
   install_dir=$(get_install_dir)
-  mkdir -p "${install_dir}/${TOOL_NAME}/${TOOL_VERSION}"
+
+  local umask=775
+  if [ "$(is_root)" -eq 0 ]; then
+    umask=755
+  fi
+
+  # shellcheck disable=SC2174
+  mkdir -p -m 775 "${install_dir}/${TOOL_NAME}"
```

Comment:
```suggestion
  mkdir -p -m "${umask}" "${install_dir}/${TOOL_NAME}"
```
So it should like this? This function is for V1 tools only?

Reply:
It is for v1  tools only, but for both - root and user - paths.
Yes, setting the umask in the first `mkdir` only would also be my suggestion. The version should be kept at 755